### DISCUSSION TITLE
Positive/Negative electrode potentials don't match actual equilibrium potentials

### DISCUSSION THREAD

**Discussion by chrisowenphd** [pybamm-team/PyBaMM#3254] [General]
Hi there, I’ve been trying to make a pseudo-half cell model for graphite by looking at the X-averaged negative electrode potential with some different parameter changes. I’ve noticed that regardless of the parameter set, the positive and negative electrode potentials don’t look like their actual half-cell equilibrium potentials. Does anyone know how to extract them? I’m trying to look at graphite staging. Below is the Prada parameter set cycled at C/10,000. The LFP electrode potential is identical to the full cell potential (should be flat) and the graphite potential just looks completely off. This holds true for other parameter sets too (Chen, Marquis, etc.): 
![Positive and Full Cell](https://github.com/pybamm-team/PyBaMM/assets/133888387/1b45e3b0-89fb-46db-b138-8af209e528a2)
![Negative](https://github.com/pybamm-team/PyBaMM/assets/133888387/baaef30e-b86c-45c0-bcea-75ebd4e3a944)

**Comment by chrisowenphd** [comment]
To show some other parameter sets and the weird negative electrode voltages at relatively higher rates, here are Chen and Marquis at C/10:

Chen:
![Chen Positive and Full Cell](https://github.com/pybamm-team/PyBaMM/assets/133888387/b7eb97e9-dbf9-4d21-bae5-cc0b86ddf7ba)
![Chen Negative](https://github.com/pybamm-team/PyBaMM/assets/133888387/67543307-6256-48a1-a47d-95b69f766915)

Marquis:
![Marquis Positive and Full Cell](https://github.com/pybamm-team/PyBaMM/assets/133888387/453d7c03-bf2e-41fa-a9c7-4444f3bb4731)
![Marquis Negative](https://github.com/pybamm-team/PyBaMM/assets/133888387/108000fb-6da7-499f-b6f8-07ef55bc3ab1)

**Reply by brosaplanella** [reply]
I am confused, aren't these voltage and current plots? Where are the potential plots for the electrodes?

**Reply by chrisowenphd** [reply]
These are plots of the individual electrode potentials (Negative and Positive) for one cycle (discharge,rest,charge). The full battery voltage is identical to the positive electrode potential. The negative electrode potential looks entirely strange. The Prada set is a good example. The LFP profile should be flat, and the graphite potential should show different stages. If you have any recommendations for pulling out the graphite voltage profile correctly, I would appreciate it.

**Reply by brosaplanella** [reply]
Can you share a minimum working example?

**Reply by rtimms** [reply]
We set the reference potential to be zero at the interface between the negative current collector and electrode, which is why the negative electrode potential is always close to zero. You can look at the OCPs of each electrode by looking at the variables "Negative electrode open circuit potential [V]" and "Positive electrode open circuit potential [V]". You can also look at the potential difference between the electrolyte and electrode at the separator interface by looking at the variables "Negative electrode surface potential difference at separator interface [V]" and "Positive electrode surface potential difference at separator interface [V]", which might be of interest too. FYI you can do `model.variables.search("string")` to search for variables.
